### PR TITLE
Add ServiceAccountAnnotations override

### LINKED ISSUE #600: Feature request: Annotations for serviceAccounts
We, as most EKS users do, rely on [IRSA roles](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/setting-up-enable-IAM.html) to grant cloud perms to K8s apps.  An `annotation` on the `serviceAccount` is required to link the running app to an IAM role.

It would be lovely to have this [serviceAccountAnnotations feature](https://github.com/DataDog/helm-charts/tree/main/charts/datadog#:~:text=agents.rbac.serviceAccountAnnotations) so EKS users can migrate from the datadog Helm chart to the excellent datadog-operator 🙏 

**Additional environment details (Operating System, Cloud provider, etc):**
AWS, EKS, K8s 1.20

### PR DESCRIPTION
**Pull request by bt-macole**
### What does this PR do?

This PR introduces `[key].ServiceAccountAnnotations` to the overrides for `nodeAgent`, `clusterAgent`, and `clusterChecksRunner`

Resolves #600

### Motivation

As noted in #600, using the DD operator has some limitations with respect to easily attaching irsa role annotations to the SA.

### Describe your test plan

`internal/controller/datadogagent/merger/rbac` and `api/datadoghq/v2alpha1/` tests were updated to validate the spec update can be read and updated following the existing conventions in the dd-operator.

> Note: was unable to pul the redhat base image so was unable to build--even before modifying the codebase--so was unable to run the full suite of tests or see final results in kubernetes.

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

**Review by alai97 — APPROVED** [review]

**Comment by alai97** [comment]
Thanks for opening this PR, @bt-macole !

**Comment by bt-macole** [comment]
> Thanks for opening this PR, @bt-macole !

Please let me know what else I can do to help get this feature added. It has been a bit of a pain point for my team and based on #600, for others as well. Would greatly appreciate any help to keep it moving forward.

File: api/datadoghq/v2alpha1/datadogagent_types.go [hunk 1]
```diff
@@ -1418,6 +1418,10 @@ type DatadogAgentComponentOverride struct {
 	// +optional
 	ServiceAccountName *string `json:"serviceAccountName,omitempty"`
 
+	// Sets the ServiceAccountAnnotations used by this component.
+	// +optional
+	ServiceAccountAnnotations map[string]string `json:"ServiceAccountAnnotations,omitempty"`
```
Comment (fanny-jiang) [inline api/datadoghq/v2alpha1/datadogagent_types.go:1423]: Nit: let's make this name lowercase to follow the convention of the other config names. 

```suggestion
	ServiceAccountAnnotations map[string]string `json:"serviceAccountAnnotations,omitempty"`
```

**Review by fanny-jiang — COMMENTED** [review]
Thanks for your contribution! I had just 1 small comment to update the config name to use lowercase camelcase.

**Review by fanny-jiang — APPROVED** [review]

**Pull request merged**